dot/core/service_integration_test.go

```diff
@@ -37,6 +38,98 @@ import (
 
 //go:generate mockgen -destination=mock_telemetry_test.go -package $GOPACKAGE github.com/ChainSafe/gossamer/dot/telemetry Client
 
+func balanceKey(t *testing.T, pub []byte) []byte {
+	h0, err := common.Twox128Hash([]byte("System"))
+	require.NoError(t, err)
+	h1, err := common.Twox128Hash([]byte("Account"))
+	require.NoError(t, err)
+	h2, err := common.Blake2b128(pub)
+	require.NoError(t, err)
+	return append(append(append(h0, h1...), h2...), pub...)
+}
+
+func newTestDigest(t *testing.T) scale.VaryingDataTypeSlice {
```

Comment:
Maybe pass in a data slice `data [][]byte`?

I'm a bit inclined to be against helping functions with actual data, it's easier to read/more explicit if the test passes the data to it.

Reply:
I recently updated it so that the data isn't hardcoded, and I think doing this would mean repeating that code for every call to this function which I think would add to much duplicated code